lib/generate-svg.js

```diff
@@ -1,12 +1,28 @@
 const fs = require('fs');
 const path = require('path');
+const svgElementAttributes = require('svg-element-attributes');
 const svgo = require('svgo');
 const merge = require('webpack-merge');
 const xmldom = require('xmldom');
 const { VAR_NAMESPACE, VAR_NAMESPACE_VALUE, hasVariables, hasVarNamespace, addVarNamespace } = require('./variable-parser');
 const generateSpritePrefix = require('./helpers/generate-sprite-prefix');
 const calculateY = require('./helpers/calculate-y');
 
+const validSymbolAttributes = [
+    ...svgElementAttributes['*'],
+    ...svgElementAttributes.svg.filter(attr => svgElementAttributes.symbol.includes(attr))
+];
+
+const validViewAttributes = [
+    ...svgElementAttributes['*'],
+    ...svgElementAttributes.svg.filter(attr => svgElementAttributes.view.includes(attr))
+];
+
+const validUseAttributes = [
+    ...svgElementAttributes['*'],
+    ...svgElementAttributes.svg.filter(attr => !['x', 'y'].includes(attr) && svgElementAttributes.use.includes(attr))
```

Comment:
What's the reason for the extra `x`/`y` check? Is it to make sure they won't be overwritten? If that's the case, maybe we should move the `attributes.forEach()` logic before the `setAttribute`s?

Reply:
Yeah that's the reason. I can move it before the setAttributes, I just wanted to maintain similar logic to the other elements, where elements are [blacklisted](https://github.com/Fire-Proof/svg-spritemap-webpack-plugin/blob/master/lib/generate-svg.js#L129,L131) and don't override existing attributes

Reply:
I agree that the logic should match for the `symbol`, `use`, and `view` elements. The blacklist is there because of SVGO and the custom parsing.